1-exercises/C-highlight-words/script.js

```diff
@@ -1,10 +1,47 @@
 function highlightWords(paragraph, colours) {
-  // Write your code here...
+  let currColor = '';
+  document.querySelector('body').style.backgroundImage = 'url(https://images.unsplash.com/photo-1529072718168-514a0d4ad1ce?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2071&q=80)';
+  document.querySelector('body').style.minHeight = '100vh';
+  document.querySelector('body').classList.add('d-flex', 'align-items-center', 'justify-content-center');
+  const container = document.getElementById('content');
+  container.classList.add('bg-light', 'text-dark', 'p-3');
+  container.style.maxWidth = '80%';
+  const select = document.createElement('select');
+  select.setAttribute('id', 'colorPicker');
+  select.classList.add('form-select-lg', 'mb-3');
+  select.addEventListener('change', (e) => {
+    currColor = e.target.value;
+  });
+
+  const optionNone = document.createElement('option');
+  optionNone.value = '';
+  optionNone.text = 'None';
+  select.appendChild(optionNone);
+
+  colours.forEach((color) => {
+    const option = document.createElement('option');
+    option.value = color;
+    option.text = color;
+    select.appendChild(option);
+  });
+  container.appendChild(select);
+  const paragraphHTML = document.createElement('p');
+  paragraphHTML.classList.add('bg-light');
+  paragraph.split(' ').forEach((word, id) => {
+    const wordHTML = document.createElement('span');
+    wordHTML.setAttribute('id', `word${id}`);
+    wordHTML.innerText = `${word} `;
+    wordHTML.addEventListener('click', () => {
+      document.getElementById(`word${id}`).style.backgroundColor = currColor;
+    });
+
+    paragraphHTML.appendChild(wordHTML);
+  });
+  container.appendChild(paragraphHTML);
```

Comment:
SImilar remark as above: better to split this code into several functions (e.g. `updateColour(event)` and `highlightWords(paragraph, colours)`)